tests/test_expression.h

```diff
@@ -0,0 +1,419 @@
+
+	CHECK_MESSAGE(
+			expression.parse("0++") == ERR_INVALID_PARAMETER,
+			"The expression shouldn't parse successfully.");
+
+	CHECK_MESSAGE(
+			expression.parse("()") == ERR_INVALID_PARAMETER,
+			"The expression shouldn't parse successfully.");
+
+	CHECK_MESSAGE(
+			expression.parse("()()") == ERR_INVALID_PARAMETER,
+			"The expression shouldn't parse successfully.");
+
+	CHECK_MESSAGE(
+			expression.parse("() - ()") == ERR_INVALID_PARAMETER,
+			"The expression shouldn't parse successfully.");
+
+	CHECK_MESSAGE(
+			expression.parse("() * 12345") == ERR_INVALID_PARAMETER,
+			"The expression shouldn't parse successfully.");
+
+	CHECK_MESSAGE(
+			expression.parse("() * 12345") == ERR_INVALID_PARAMETER,
+			"The expression shouldn't parse successfully.");
+
+	CHECK_MESSAGE(
+			expression.parse("123'456") == ERR_INVALID_PARAMETER,
+			"The expression shouldn't parse successfully.");
+
+	CHECK_MESSAGE(
+			expression.parse("123\"456") == ERR_INVALID_PARAMETER,
+			"The expression shouldn't parse successfully.");
+}
+
+TEST_CASE("[Expression] Unusual expressions") {
+	Expression expression;
+
+	// Redundant parentheses don't cause a parse error as long as they're matched.
+	CHECK_MESSAGE(
+			expression.parse("(((((((((((((((666)))))))))))))))") == OK,
+			"The expression should parse successfully.");
+
+	// Using invalid identifiers doesn't cause a parse error.
+	ERR_PRINT_OFF;
+	CHECK_MESSAGE(
+			expression.parse("hello + hello") == OK,
+			"The expression should parse successfully.");
+	CHECK_MESSAGE(
+			int(expression.execute()) == 0,
+			"The expression should return the expected result.");
+	ERR_PRINT_ON;
+
+	ERR_PRINT_OFF;
+	CHECK_MESSAGE(
+			expression.parse("$1.00 + €5") == OK,
+			"The expression should parse successfully.");
+	CHECK_MESSAGE(
+			int(expression.execute()) == 0,
+			"The expression should return the expected result.");
+	ERR_PRINT_ON;
+
+	// Commas can't be used as a decimal parameter.
+	CHECK_MESSAGE(
+			expression.parse("123,456") == OK,
+			"The expression should parse successfully.");
+	CHECK_MESSAGE(
+			int(expression.execute()) == 123,
+			"The expression should return the expected result.");
+
+	// Spaces can't be used as a separator for large numbers.
+	CHECK_MESSAGE(
+			expression.parse("123 456") == OK,
+			"The expression should parse successfully.");
+	CHECK_MESSAGE(
+			int(expression.execute()) == 123,
+			"The expression should return the expected result.");
+
+	// Division by zero is accepted, even though it prints an error message normally.
+	CHECK_MESSAGE(
+			expression.parse("-25.4 / 0") == OK,
+			"The expression should parse successfully.");
+	ERR_PRINT_OFF;
+	CHECK_MESSAGE(
+			Math::is_zero_approx(float(expression.execute())),
+			"`-25.4 / 0` should return 0.");
+	ERR_PRINT_ON;
+
+	CHECK_MESSAGE(
+			expression.parse("0 / 0") == OK,
+			"The expression should parse successfully.");
+	ERR_PRINT_OFF;
+	CHECK_MESSAGE(
+			int(expression.execute()) == 0,
+			"`0 / 0` should return 0.");
```

Comment:
Following fun expressions should be added to tests as well:
```
 (-9223372036854775807 - 1) % -1
 (-9223372036854775807 - 1) / -1
```

Both are still [causing crash](https://github.com/godotengine/godot/issues/26131) in `Expression` and `GDScript`.

Reply:
Another strange behavior: `-9223372036854775808` (`INT64_MIN`) can't be used in expressing and result in `ERROR: Cannot represent 9223372036854775808 as 64-bit integer, provided value is too big.`, I guess it's reading value as positive first and then negates it.

Reply:
I added it as commented out code, since it crashes the engine when run.